### ISSUE TITLE
HTML UVOZ

### ISSUE THREAD

**Issue by evawinkler** [evawinkler/APPR-2019-20#4]
Zanima me ali je poleg uvoza podatkov v csv obliki, potrebujem uvoz v html? 
Poleg tega sem po zgledu iz vaj želela uvoziti tabelo iz spletne strani: [https://en.wikipedia.org/wiki/Economy_of_Slovenia](url)

`link <- "https://en.wikipedia.org/wiki/Economy_of_Slovenia"
stran <- html_session(link) %>% read_html()
tabela <- stran %>% html_nodes(xpath="//table[@class='wikitable sortable']") %>%
  .[[1]] %>% html_table(dec=".")

colnames(tabela) <- c("leto", "1993", "1995", "2000", 2005 : 2017)

for (col in c( "1993", "1995", "2000", 2005 : 2017)) {
  tabela[[col]] <- parse_number(tabela[[col]], na="-", locale=sl)
}

for (col in c("leto")) {
  tabela[[col]] <- factor(tabela[[col]])
}`

A ko to zaženem se mi pokaže problem `Error in .[[1]] : subscript out of bounds`
Zanima me kako lahko ta problem rešim? 
Hvala za odgovor že v naprej

**Comment by timotejvesel** [comment]
Ja, nekaj podatkov (npr. eno tabelo) naj bi imeli uvoženih tudi iz oblike html. 
Zgornjo tabelo uvoziš kot
```r
link <- "https://en.wikipedia.org/wiki/Economy_of_Slovenia"
stran <- html_session(link) %>% read_html()
tabela <- stran %>% html_nodes(xpath="//table[@class='wikitable']") %>%
  .[[1]] %>% html_table(dec=".")
```
Torej `@Class='wikitable sortable'` zamenjaš z `@class='wikitable'`.

**Comment by evawinkler** [comment]
Najlepša hvala, mi je uvozilo tabelo. 
Zanima me še , kako naj spremenim vrstice v tabeli, saj bi rada da so podatki v %? 
Ko sem uvozila  html tabelo mi je spremenilo zapis nekaterih podatkov, npr. namesto 8,6 % v 86.000. 
Ali lahko problem rešim tako da delim podatke s 10, ali obstaja kakšen drugačen način? 
hvala v naprej

**Comment by evawinkler** [comment]
Sem ugotovila, hvala. Najprej sem obdržala kar svojo pa samo spremenila @class='wikitable', pa zato mi je spremenilo številke. Sedaj sem napisala pa vašo funkcijo, ki ste mi jo zgoraj napisali, in deluje pravilno.

**Comment by timotejvesel** [comment]
Pri pretvorbi vrednosti iz `character` v `double` je še vedno nekaj problemov. Problem je znak *−*, ki je drugačen od znaka za minus *-*, zato ga pri pretvorbi izbriše (torej negativno število spremeni v pozitivno), Problem je tudi vejica (v 2.vrstici), ki jo zazna kot decimalni znak čeprav to ni.  
S tem bi morale biti te težave odpravljene:
```r
link <- "https://en.wikipedia.org/wiki/Economy_of_Slovenia"
stran <- html_session(link) %>% read_html()
tabela <- stran %>% html_nodes(xpath="//table[@class='wikitable']") %>%
  .[[1]] %>% html_table(dec=".")

colnames(tabela) <- c("leto", "1993", "1995", "2000", 2005 : 2017)

tabela[] <- lapply(tabela, gsub, pattern="%", replacement="")
tabela[] <- lapply(tabela, gsub, pattern="−", replacement="-")
tabela[] <- lapply(tabela, gsub, pattern=",", replacement="")

for (col in c( "1993", "1995", "2000", 2005 : 2017)) {
  tabela[[col]] <- parse_number(tabela[[col]], na="...")
}

tabela[1,1] <- "GDP in €(PPP) v milijardah" 
```

**Comment by evawinkler** [comment]
Najlepša hvala, sedaj deluje vse .
